src/index.ts

```diff
@@ -61,9 +61,9 @@ export class Pipeline {
     }
 
     @Expose({ name: 'steps' })
-    private _steps(): (WaitStep | Step)[] {
+    private async _steps(): Promise<(WaitStep | Step)[]> {
```

Comment:
this one serializes to `{}` due to promises not being supported by https://github.com/typestack/class-transformer